rest/route/service.go

```diff
@@ -61,10 +61,10 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddWrapper(gimlet.WrapperMiddleware(allowCORS))
 
 	// Agent routes
-	app.AddRoute("/pods/{pod_id}/agent/next_task").Version(2).Get().Wrap(requirePod).RouteHandler(makePodAgentNextTask(env))
-	app.AddRoute("/pods/{pod_id}/task/{task_id}/end").Version(2).Post().Wrap(requirePod, requireTask).RouteHandler(makePodAgentEndTask(env))
-	app.AddRoute("/hosts/{host_id}/agent/next_task").Version(2).Get().Wrap(requireHost).RouteHandler(makeHostAgentNextTask(env, opts.TaskDispatcher, opts.TaskAliasDispatcher))
-	app.AddRoute("/hosts/{host_id}/task/{task_id}/end").Version(2).Post().Wrap(requireHost, requireTask).RouteHandler(makeHostAgentEndTask(env))
+	app.AddRoute("/pods/{pod_id}/agent/next_task").Version(2).Get().Wrap(requirePodOrHost).RouteHandler(makePodAgentNextTask(env))
```

Comment:
Rather than change all the pod or host specific ones to accept both auth methods, it should be safe to change `requireHost` to [do the same logic as pod or host auth for hosts](https://github.com/evergreen-ci/evergreen/blob/b28150c48556bf161354f9b80990f5d9eb36af82/rest/route/middleware.go#L345-L358). I checked the two routes that use it and both of them are agent routes where it's okay to update the last communicated time.

Reply:
Sounds good, I've moved this to a helper function